plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -299,12 +377,25 @@ public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull Style
                     }
                     applyEditorStyle();
                 }
-            });
-        } catch (Exception e) {
-            throw new DBException("Error loading text value", e);
-        } finally {
-            monitor.done();
-        }
+            }
+        });
+    }
+
+    private void showRestrictedContent(@NotNull DBDContent value) {
+        int maxContentSize = DBWorkbench.getPlatform().getPreferenceStore()
+            .getInt(ModelPreferences.REPRESENTATION_CONTENT_MAX_SIZE_KBYTES);
+        byte[] bytes = CommonUtils.toString(value).getBytes();
+        byte[] restrictedBytes = Arrays.copyOfRange(bytes, 0, maxContentSize);
```

Comment:
What issues with zero at the end? The restricted content is cutting and is in read only mode, auto formating aslo does not work for this case.

Reply:
I've shown you the issue with SQLite. It's caused by trailing zeroes.

Reply:
Should not happen because of 
```
if (value.getContentLength() > maxContentSize) {
                showRestrictedContent(value);
            } else {
                showRegularContent(monitor);
            }
```

Reply:
But it's happening. Also content conversion is off in SQLite (it should display trimmed data, not the displayed placeholder):

|Without my suggestion|With my suggestion|
|---|---|
|![java_mpO8C4cO13](https://github.com/dbeaver/dbeaver/assets/35821147/2936ddd5-a40e-4b0a-9dc3-27c170f6db1e)|![ZZPAk7FhiL](https://github.com/dbeaver/dbeaver/assets/35821147/72f5ca95-b7af-4951-a73d-e470182e54a2)|

Reply:
Content restricted by 1000 KB, but the right part looks a 100 bytes, not sure is it correct.

Reply:
You're basically doing `content.toString()`, which is incorrect. You should use `DBDContent#getContents` instead.

Reply:
This is a bug, and I won't pass this PR to the QA team until it's fixed.

Reply:
Let's better use DBDContent#getContents. No need to call toString when we have API for that needs.